portal/app/farm_updater/egauge_api.py

```diff
@@ -92,12 +92,19 @@ def processing_egauge_data(self):
         # taking only the first Eguage -> Need to automate this in case there are more egauges
         data_queryset = FarmData.objects.filter(farm_device=queryset[0]).order_by('-id')
 
+        # Checking if there's data stored already
         if not data_queryset:
             print('No previous Egauge data exists...creating first entry')
             return json.dumps(egauge_data)
 
         data_prev = json.loads(data_queryset[0].device_data)['raw']
 
+        # Checking if timestamp is different than null
+        if data_prev is None:
+            print('Problem saving previous egauge data')
+            return json.dump(egauge_data)
```

Comment:
This still doesn't fix the issue. You'll still end up with junk data in the DB.

Reply:
There's a bunch of code paths that simply return json.dumps(...) w/o consideration of whether the data structure is correct. You should add a check fo whether the response contains the expected data and if not, don't save. So, right now, if the json is valid (even if it's an error) you'll save it in the DB. Which may cause an error in the future.

Reply:
I'm returning None whenever there is an error getting data from egauge. This return will make sure nothing incorrect is saved in the DB. Not sure if this is the best approach though.